pyartifactory/models/artifact.py

```diff
@@ -66,6 +66,30 @@ class ArtifactFileInfoResponse(ArtifactInfoResponseBase):
     originalChecksums: Optional[OriginalChecksums] = None
 
 
+class ArtifactFolderListResponse(BaseModel):
+    """Models an artifact folder list response."""
+
+    uri: str
+    size: float
+    lastModified: datetime
+    folder: bool
+
+
+class ArtifactFileListResponse(BaseModel):
+    """Models an artifact file list response."""
+
+    sha1: Optional[str] = None
+    sha2: Optional[str] = None
+
+
+class ArtifactListResponse(BaseModel):
+    """Models an artifact list response."""
+
+    uri: str
+    created: datetime
+    files: List[Union[ArtifactFileListResponse, ArtifactFolderListResponse]]
```

Comment:
I think while developing the models you forgot to make them inherit each other.
As we see in the example response in the docs, a file and a folder share mostly the same attributes (uri, size, lastModified and folder). You correctly reproduced this schema with the `ArtifactFolderListResponse` but the `ArtifactFileListResponse` lacks all of them. This also means that when parsing the response, every JSON object can and will be matched to a `ArtifactFileListResponse`, as its only attributes are optional sha1 and sha2.

I fixed this in the following suggestion: a common class `ArtifactListEntryResponse` has the common attributes, then `ArtifactListFolderResponse` and `ArtifactListFileResponse` both inherit it, while also redefining the `folder` attribute to be literally `True` or `False` (note that this requires a `from typing import Literal` to be added at the top of the file).
When parsing, Pydantic will try to match any JSON object to one of the class, with this `folder` attribute acting as a discriminator.
Note that I also changed the `size` attribute to be an `int` instead of a `float`, since the API returns bytes it should never be fractional.

```suggestion
# Add a `from typing import Literal` at the top of the file

class ArtifactListEntryResponse(BaseModel):
    """Base model for an entry in an artifact list response."""

    uri: str
    size: int
    lastModified: datetime
    folder: bool


class ArtifactListFolderResponse(ArtifactListEntryResponse):
    """Models a folder in an artifact list response."""

    folder: Literal[True]


class ArtifactListFileResponse(ArtifactListEntryResponse):
    """Models a file in an artifact list response."""

    folder: Literal[False]
    sha1: Optional[str] = None
    sha2: Optional[str] = None


class ArtifactListResponse(BaseModel):
    """Models an artifact list response."""

    uri: str
    created: datetime
    files: List[Union[ArtifactListFileResponse, ArtifactListFolderResponse]]
```

Reply:
@nymous Thanks for the good eye, something I would have picked up while unit testing.
I will implement the changes.